### PR TITLE
Suggested route by vessel report created

### PR DESCRIPTION
**Pull request by fawadik**
Endpoint created: Get suggested route by Vessel report

### CODE REVIEW

**Review by AbodHassam — DISMISSED** [review]
Looks good to me

File: src/server/api/controllers/vessel-reports.controller.js [hunk 1]
```diff
@@ -74,8 +74,28 @@ const getVesselReportById = async id => {
   }
 };
 
+//Get Suggested-route by vessel-report-id
+const getsuggestedRouteByVesselReportId = async id => {
```
Comment (Swapna-Sahu) [inline src/server/api/controllers/vessel-reports.controller.js]: Looks good to me but only one comment. Can you rename the function name to "getSuggestedRouteByVesselReportId"

File: src/server/api/controllers/vessel-reports.controller.js [hunk 2]
```diff
@@ -74,8 +74,28 @@ const getVesselReportById = async id => {
...
+    return err.message;
+  }
+};
+
 module.exports = {
   createVesselReport,
   getVesselReportById,
-  getVesselsReportByVoyageId
+  getVesselsReportByVoyageId,
+  getsuggestedRouteByVesselReportId
```
Comment (Swapna-Sahu) [inline src/server/api/controllers/vessel-reports.controller.js]: Can you rename the function name to "getSuggestedRouteByVesselReportId"

File: src/server/api/routes/vessel-reports.router.js [hunk 3]
```diff
@@ -31,4 +31,16 @@ router.get("/:id", authorizeUser(ROLES.SUPER_USER), function(req, res, next) {
     .catch(next);
 });
 
+// ENDPOINT: /api/vessel-reports/:vessel-report-id/suggested-route
+router.get(
+  "/:vessel_report_id/suggested-route",
+  authorizeUser(ROLES.SUPER_USER),
+  function(req, res, next) {
+    vesselReportsController
+      .getsuggestedRouteByVesselReportId(req.params.vessel_report_id)
```
Comment (Swapna-Sahu) [inline src/server/api/routes/vessel-reports.router.js]: Can you rename the function name to "getSuggestedRouteByVesselReportId"

**Review by Swapna-Sahu — CHANGES_REQUESTED** [review]
All good only need one modification

File: src/server/api/routes/vessel-reports.router.js [hunk 4]
```diff
@@ -31,4 +31,16 @@ router.get("/:id", authorizeUser(ROLES.SUPER_USER), function(req, res, next) {
     .catch(next);
 });
 
+// ENDPOINT: /api/vessel-reports/:vessel-report-id/suggested-route
+router.get(
+  "/:vessel_report_id/suggested-route",
```
Comment (zkwsk) [inline src/server/api/routes/vessel-reports.router.js]: Suggested routes should be plural `suggested_routes` to indicate that we are getting _all_ suggested_routes matching a given `vessel_id`.

File: src/server/api/controllers/vessel-reports.controller.js (hunk 1 above)
Comment (zkwsk) [reply src/server/api/controllers/vessel-reports.controller.js]: Agree. Please fix this.

**Review by zkwsk — CHANGES_REQUESTED** [review]
It works and looks good overall.

Can you please fix the small comments and then add documentation to swagger.json and make sure that you can call the endpoint from swagger?

File: src/server/api/controllers/vessel-reports.controller.js (hunk 2 above)
Comment (fawadik) [reply src/server/api/controllers/vessel-reports.controller.js]: changed

File: src/server/api/routes/vessel-reports.router.js (hunk 3 above)
Comment (fawadik) [reply src/server/api/routes/vessel-reports.router.js]: changed

File: src/server/api/routes/vessel-reports.router.js (hunk 4 above)
Comment (fawadik) [reply src/server/api/routes/vessel-reports.router.js]: made it plural

File: src/server/api/controllers/vessel-reports.controller.js (hunk 1 above)
Comment (fawadik) [reply src/server/api/controllers/vessel-reports.controller.js]: fixed

**Review by Swapna-Sahu — DISMISSED** [review]
Looks good to me

File: src/server/api/controllers/vessel-reports.controller.js [hunk 5]
```diff
@@ -74,8 +74,28 @@ const getVesselReportById = async id => {
   }
 };
 
+//Get Suggested-routes by vessel-report-id
```
Comment (zkwsk) [inline src/server/api/controllers/vessel-reports.controller.js]: This function should be in the `suggested-routes.route.js`

**Review by zkwsk — CHANGES_REQUESTED** [review]

**Review by Swapna-Sahu — DISMISSED** [review]
looks good to me

**Review by zkwsk — DISMISSED** [review]
Good work 🥇

**Review by AbodHassam — APPROVED** [review]
Looks good to me 👍

**Review by zkwsk — APPROVED** [review]
Very nice 💯

**Pull request merged**